src/base/bittorrent/customstorage.cpp

```diff
@@ -35,6 +35,191 @@
 #include "base/utils/fs.h"
 #include "common.h"
 
+#if (LIBTORRENT_VERSION_NUM >= 20000)
+#include <libtorrent/session.hpp>
+
+std::unique_ptr<lt::disk_interface> customDiskIOConstructor(
+        lt::io_context &ioContext, const lt::settings_interface &settings, lt::counters &counters)
+{
+    return std::make_unique<customDiskIOThread>(lt::default_disk_io_constructor(ioContext, settings, counters));
+}
+
+customDiskIOThread::customDiskIOThread(std::unique_ptr<libtorrent::disk_interface> nativeDiskIOThread)
+    : m_nativeDiskIO {std::move(nativeDiskIOThread)}
+{
+}
+
+lt::storage_holder customDiskIOThread::new_torrent(const lt::storage_params &storageParams, const std::shared_ptr<void> &torrent)
+{
+    lt::storage_holder storageHolder = m_nativeDiskIO->new_torrent(storageParams, torrent);
```

Comment:
maybe this variable can become `const`?

Reply:
I think there's a subtle benefit of it being mutable. it allows it to be *moved* out when returned (which is implicit), unless of course the guaranteed copy elision kicks in here, I don't recall all the conditions for that.

Reply:
Personally I think making it immutable helps when reading the code and it outweigh the penalty at non-performance critical places.
https://www.bfilipek.com/2017/01/const-move-and-rvo.html

Reply:
OK.